.github/workflows/system-tests-opensource.yml

```diff
@@ -55,165 +56,215 @@ on:
+          echo "mlrun_hash=$mlrun_hash" >> $GITHUB_OUTPUT
+          echo "ui_hash=$ui_hash" >> $GITHUB_OUTPUT
+
+          echo "mlrun_version=$(echo ${{ steps.git_upstream_info.outputs.unstable_version_prefix }}+$mlrun_hash)" >> $GITHUB_OUTPUT
+          echo "mlrun_ui_version=${{ steps.git_upstream_info.outputs.unstable_version_prefix }}-$ui_hash" >> $GITHUB_OUTPUT
+          echo "mlrun_docker_tag=$(echo ${{ steps.git_upstream_info.outputs.unstable_version_prefix }}-$mlrun_hash)" >> $GITHUB_OUTPUT
+          
+          export mlrun_version_specifier=$mlrun_hash
+          if [ -n "$PR_NUMBER" ]; then
+            mlrun_version_specifier="refs/pull/${PR_NUMBER}/merge"
+          fi
+          echo "mlrun_version_specifier=$mlrun_version_specifier" >> $GITHUB_OUTPUT
+          
+          echo "mlrun_docker_registry=ghcr.io/" >> $GITHUB_OUTPUT
+          echo "mlrun_docker_repo=$(echo ${{ github.repository_owner }} | tr '[:upper:]' '[:lower:]')" >> $GITHUB_OUTPUT
+          
+          echo "mlrun_system_tests_clean_resources=$( \
+            input_system_tests_clean_resources=$INPUT_CLEAN_RESOURCES_IN_TEARDOWN && \
+            echo ${input_system_tests_clean_resources:-true})" >> $GITHUB_OUTPUT
+        env:
+          INPUT_CLEAN_RESOURCES_IN_TEARDOWN: ${{ github.event.inputs.clean_resources_in_teardown }}
+          PR_NUMBER: ${{ github.event.inputs.pr_number }}
 
-    # let's not run this on every fork, change to your fork when developing
-    if: github.repository == 'mlrun/mlrun' || github.event_name == 'workflow_dispatch'
+  build-mlrun:
+    if : github.event.inputs.pr_number != ''
+    name: Build MLRun
+    uses: ./.github/workflows/build-internal.yaml
+    needs: prepare-inputs
+    with:
+      docker_registries: ${{ needs.prepare-inputs.outputs.mlrun_docker_registry }}
+      docker_repo: ${{ needs.prepare-inputs.outputs.mlrun_docker_repo }}
+      version: ${{ needs.prepare-inputs.outputs.mlrun_version }}
+      cache_tag_suffix:  ${{ needs.prepare-inputs.outputs.mlrun_docker_tag }}
 
+      # NOTE: currently we patch mlrun/api, mlrun/log-collector and mlrun/ui images.
+      # by overriding the version specifier, we allow using mlrun/mlrun as well.
+      # TODO: support patching jupyter as well (maybe on demand?)
+      # and use --override-mlrun-jupyter-image when patching mlrun ce
+      skip_images: test,mlrun-gpu,base,jupyter
+    secrets: inherit
+
+  run-system-tests-opensource-ci:
+    name: Run System Tests Open Source
+    runs-on: ubuntu-latest
+    needs: [prepare-inputs, build-mlrun]
     steps:
     - uses: actions/checkout@v4
+
     # since github-actions gives us 14G only, and fills it up with some garbage
     - name: Freeing up disk space
       run: |
         "${GITHUB_WORKSPACE}/automation/scripts/github_workflow_free_space.sh"
-    - name: Set up python
-      uses: actions/setup-python@v5
+
+    - name: Install uv
+      uses: astral-sh/setup-uv@v3
```

Comment:
I think if you setup python and then setup uv then you don't need to `source venv/bin/activate` on each step
https://github.com/charliermarsh/autobot/blob/e42c66659bf97b90ca9ff305a19cc99952d0d43f/.github/workflows/ci.yaml
Not sure about it but worth checking

Reply:
note that they run uv run python so it is basically like sourcing the python venv

the reason I had to install python and venv is because I can't use system python but rather the venv one